lib/DxilContainer/DxcContainerBuilder.cpp

```diff
@@ -164,11 +169,62 @@ DxcContainerBuilder::SerializeContainer(IDxcOperationResult **ppResult) {
         {DxcOutputObject::DataOutput(DXC_OUT_OBJECT, pResult, DxcOutNoName),
          DxcOutputObject::DataOutput(DXC_OUT_ERRORS, pErrorBlob, DxcOutNoName)},
         ppResult));
+
+    // Add Hash.
+    if (ppResult != nullptr && *ppResult != nullptr) {
+      HRESULT HR;
+      (*ppResult)->GetStatus(&HR);
+      if (SUCCEEDED(HR)) {
+        CComPtr<IDxcBlob> pObject;
+        HR = (*ppResult)->GetResult(&pObject);
+        if (SUCCEEDED(HR)) {
+          LPVOID PTR = pObject->GetBufferPointer();
+          if (IsDxilContainerLike(PTR, pObject->GetBufferSize()))
+            HashAndUpdate((DxilContainerHeader *)PTR);
+        }
+      }
+    }
     return S_OK;
   }
   CATCH_CPP_RETURN_HRESULT();
 }
 
+void DxcContainerBuilder::FindHashFunctionFromSource(
```

Comment:
I'm a fan of descriptive comments above any function, but this one in particular seems to need it. 

```suggestion
// Try hashing the source contained in ContainerHeader using retail and debug
// hashing functions. If either of them match the stored result, set the
// HashFunction to the matching variant. If neither match, set it to null.
void DxcContainerBuilder::FindHashFunctionFromSource(
```

Reply:
Updated.